Doc: Update AVD 5.7 release notes to deprecate the keys removed in AVD 6.0 (#6439)
aristanetworks/ansible-avd: docs/release-notes/5.x.x.md

- `arista.avd.eos_validate_state_report`
- `arista.avd.eos_validate_state_runner`

#### Deprecations on eos_designs data models

> The following data model keys have been deprecated from eos_designs in AVD 5.7.0 and will be removed in AVD 6.0.0.

Reviewer: ```suggestion
The following data model keys have been deprecated from eos_designs in AVD 5.7.3 and will be removed in AVD 6.0.0.
```

| Deprecated key | New key in AVD 6.0 | Notes for AVD 6.0 |
| -------------- | ------------------ | ----------------- |
| `avd_data_validation_mode` | - | Validation is always required. |
| `avd_eos_designs_debug` | avd_tmp_dir | AVD will output internal temporary files to `avd_tmp_dir` containing validated variables and facts. |
| `avd_eos_designs_enforce_duplication_checks_across_all_models` | - | This is the builtin behavior. |
| `avd_eos_designs_unset_facts` | - | `avd_switch_facts` are no longer set in Ansible. Can be read by setting `avd_tmp_dir`. |
| `wan_use_agent_env_var_for_kernel_software_forwarding_ecmp` | `<node_type>.kernel_ecmp_cli` | Set new key to `false` to retain the behavior. |
| `wan_use_evpn_node_settings_for_lan` | - | Default in AVD 6.0 |

!!! note
    When upgrading to AVD 6.0, refer to the porting-guide for all required changes.

#### Deprecations on eos_cli_config_gen data models

> The following data model keys have been deprecated from eos_cli_config_gen in AVD 5.7.0 and will be removed in AVD 6.0.0.

Reviewer: ```suggestion
The following data model keys have been deprecated from eos_cli_config_gen in AVD 5.7.3 and will be removed in AVD 6.0.0.
```

| `is_deployed` | `metadata.is_deployed` | Only used for validation and documentation purposes. |
| `ntp.servers[].vrf` | `ntp.vrf` | EOS supports one VRF for all NTP servers. |
| `port_channel_interfaces[].peer` | `port_channel_interfaces[].metadata.peer` | Only used for validation and documentation purposes. |
| `port_channel_interfaces[].peer_interface` | `port_channel_interfaces[].metadata.peer_interface` | Only used for validation and documentation purposes. |
| `port_channel_interfaces[].peer_key` | `port_channel_interfaces[].metadata.peer_key` | Only used for validation and documentation purposes. |
| `port_channel_interfaces[].peer_type` | `port_channel_interfaces[].metadata.peer_type` | Only used for validation and documentation purposes. |
| `port_channel_interfaces[].validate_lldp` | `port_channel_interfaces[].metadata.validate_lldp` | Only used for validation and documentation purposes. |
| `port_channel_interfaces[].validate_state` | `port_channel_interfaces[].metadata.validate_state` | Only used for validation and documentation purposes. |

> | `radius_server.hosts` | `radius_server.servers` | Used to configure hosts with default VRF |

Reviewer 2: ```suggestion
| `radius_server.hosts` | `radius_server.servers` or `radius_server.vrfs[].servers` | Improved input data model for CLI ordering. |
```

For more details, see the `arista.avd.cv_deploy` [role documentation](../../ansible_collections/arista/avd/roles/cv_deploy/README.md).

#### Deprecations on eos_designs data-models

The following data model keys have been deprecated from eos_designs in AVD 5.7.0 and will be removed in AVD 6.0.0.

| Deprecated key | New key in AVD 6.0 |
| -------------- | ------------------ |
| avd_data_validation_mode | - |
| avd_eos_designs_debug | - |
| avd_eos_designs_enforce_duplication_checks_across_all_models | - |
| avd_eos_designs_unset_facts | - |

> | wan_use_agent_env_var_for_kernel_software_forwarding_ecmp | <node_type>.kernel_ecmp_cli |

Reviewer 2: ```suggestion
| Deprecated key | New key in AVD 6.0 | Notes for AVD 6.0 |
| -------------- | ------------------ | ----------------- |
| avd_data_validation_mode | - | Validation is always required.
| avd_eos_designs_debug | avd_tmp_dir | AVD will output internal temporary files to `avd_tmp_dir` containing validated variables and facts. |
| avd_eos_designs_enforce_duplication_checks_across_all_models | - | This is the builtin behavior. |
| avd_eos_designs_unset_facts | - | `avd_switch_facts` are no longer set in Ansible. Can be read by setting `avd_tmp_dir`. |
| wan_use_agent_env_var_for_kernel_software_forwarding_ecmp | <node_type>.kernel_ecmp_cli | |
```

#### Deprecations on eos_cli_config_gen data-models

The following data model keys have been deprecated from eos_cli_config_gen in AVD 5.7.0 and will be removed in AVD 6.0.0.

| Deprecated key | New key in AVD 6.0 |
| -------------- | ------------------ |
| avd_data_validation_mode | - |
| dhcp_servers[].subnets | dhcp_servers[].ipv4_subnets or dhcp_servers[].ipv6_subnets |
| ip_ftp_client_source_interfaces | ip_ftp_client |
| ip_http_client_source_interfaces | ip_http_client |
| ip_name_servers | ip_name_server |
| ip_ssh_client_source_interfaces| ip_ssh_client |
| ip_telnet_client_source_interfaces | ip_telnet_client |
| ip_tftp_client_source_interfaces | ip_tftp_client |

> | ntp.servers[].vrf | ntp.vrf |

Reviewer 2: ```suggestion
| Deprecated key | New key in AVD 6.0 | Notes for AVD 6.0 |
| -------------- | ------------------ | ----------------- |
| avd_data_validation_mode | - | Validation is always required. |
| dhcp_servers[].subnets | dhcp_servers[].ipv4_subnets or dhcp_servers[].ipv6_subnets | |
| ip_ftp_client_source_interfaces | ip_ftp_client.source_interface or ip_ftp_client.vrfs[].source_interface | |
| ip_http_client_source_interfaces | ip_http_client.source_interface or ip_http_client.vrfs[].source_interface | |
| ip_name_servers | ip_name_server.servers |
| ip_ssh_client_source_interfaces| ip_ssh_client.source_interface or ip_ssh_client.vrfs[].source_interface | |
| ip_telnet_client_source_interfaces | ip_telnet_client.source_interface or ip_telnet_client.vrfs[].source_interface | |
| ip_tftp_client_source_interfaces | ip_tftp_client.source_interface or ip_tftp_client.vrfs[].source_interface | |
| ntp.servers[].vrf | ntp.vrf | EOS supports one VRF for all NTP servers. |
```

For more details, see the `arista.avd.cv_deploy` [role documentation](../../ansible_collections/arista/avd/roles/cv_deploy/README.md).

#### Deprecations on eos_designs data-models

The following data model keys have been deprecated from eos_designs in AVD 5.7.0 and will be removed in AVD 6.0.0.

| Deprecated key | New key in AVD 6.0 | Notes for AVD 6.0 |
| -------------- | ------------------ | ----------------- |
| avd_data_validation_mode | - | Validation is always required. |
| avd_eos_designs_debug | avd_tmp_dir | AVD will output internal temporary files to `avd_tmp_dir` containing validated variables and facts. |
| avd_eos_designs_enforce_duplication_checks_across_all_models | - | This is the builtin behavior. |
| avd_eos_designs_unset_facts | - | `avd_switch_facts` are no longer set in Ansible. Can be read by setting `avd_tmp_dir`. |

> | evpn_ebgp_gateway_inter_domain | - | - |

Author: ```suggestion
| evpn_ebgp_gateway_inter_domain | - | Removing unused key. |
```

> | `wan_use_agent_env_var_for_kernel_software_forwarding_ecmp` | `<node_type>.kernel_ecmp_cli` | - |

Reviewer: ```suggestion
| `wan_use_agent_env_var_for_kernel_software_forwarding_ecmp` | `<node_type>.kernel_ecmp_cli` | Set new key to `false` to retain the behavior. |
```

| Deprecated key | New key in AVD 6.0 | Notes for AVD 6.0 |
| -------------- | ------------------ | ----------------- |
| avd_data_validation_mode | - | Validation is always required. |
| dhcp_servers[].subnets | dhcp_servers[].ipv4_subnets or dhcp_servers[].ipv6_subnets | - |
| ethernet_interfaces[].peer | ethernet_interfaces[].metadata.peer | Only used for validation and documentation purposes |
| ethernet_interfaces[].peer_interface | ethernet_interfaces[].metadata.peer_interface | Only used for validation and documentation purposes |
| ethernet_interfaces[].peer_key | ethernet_interfaces[].metadata.peer_key | Only used for validation and documentation purposes |
| ethernet_interfaces[].peer_type | ethernet_interfaces[].metadata.peer_type | Only used for validation and documentation purposes |
| ethernet_interfaces[].port_porfile | ethernet_interfaces[].metadata.port_porfile | Only used for validation and documentation purposes |
| ethernet_interfaces[].validate_lldp | ethernet_interfaces[].metadata.validate_lldp | Only used for validation and documentation purposes |

> | ethernet_interfaces[].validate_state | ethernet_interfaces[].metadata.validate_state | Only used for validation and documentation purposes |

Author: ```suggestion
| ethernet_interfaces[].peer | ethernet_interfaces[].metadata.peer | Only used for validation and documentation purposes. |
| ethernet_interfaces[].peer_interface | ethernet_interfaces[].metadata.peer_interface | Only used for validation and documentation purposes. |
| ethernet_interfaces[].peer_key | ethernet_interfaces[].metadata.peer_key | Only used for validation and documentation purposes. |
| ethernet_interfaces[].peer_type | ethernet_interfaces[].metadata.peer_type | Only used for validation and documentation purposes. |
| ethernet_interfaces[].port_porfile | ethernet_interfaces[].metadata.port_porfile | Only used for validation and documentation purposes. |
| ethernet_interfaces[].validate_lldp | ethernet_interfaces[].metadata.validate_lldp | Only used for validation and documentation purposes. |
| ethernet_interfaces[].validate_state | ethernet_interfaces[].metadata.validate_state | Only used for validation and documentation purposes. |
```

| router_bgp.vlan_aware_bundles[].description | router_bgp.vlan_aware_bundles[].metadata.description | Only used for validation and documentation purposes |
| router_bgp.vlan_aware_bundles.tenant | router_bgp.vlan_aware_bundles.metadata.tenant | Only used for validation and documentation purposes |
| router_bgp.vlans.tenant | router_bgp.vlans.metadata.tenant | Only used for validation and documentation purposes |
| serial_number | metadata.serial_number | Only used for validation and documentation purposes |
| vlan_interfaces.ipv6_ospf_area | vlan_interfaces[].ipv6_ospf.process.area | - |
| vlan_interfaces[].ipv6_ospf_network_point_to_point | vlan_interfaces[].ipv6_ospf.network_point_to_point | - |
| vlan_interfaces[].tags | vlan_interfaces[].metadata.tags | Only used for validation and documentation purposes |
| vlan_interfaces[].tenant | vlan_interfaces[].metadata.tenant | Only used for validation and documentation purposes |
| vlan_interfaces[].type | vlan_interfaces[].metadata.type | Only used for validation and documentation purposes |
| vlans[].tenant | vlans[].metadata.tenant | Only used for validation and documentation purposes |

> | vrfs[].tenant | vrfs[].metadata.tenant | Only used for validation and documentation purposes |

Author: ```suggestion
| is_deployed | metadata.is_deployed | Only used for validation and documentation purposes. |
| ntp.servers[].vrf | ntp.vrf | EOS supports one VRF for all NTP servers. |
| port_channel_interfaces[].peer | port_channel_interfaces[].metadata.peer | Only used for validation and documentation purposes. |
| port_channel_interfaces[].peer_interface | port_channel_interfaces[].metadata.peer_interface | Only used for validation and documentation purposes. |
| port_channel_interfaces[].peer_key | port_channel_interfaces[].metadata.peer_key | Only used for validation and documentation purposes. |
| port_channel_interfaces[].peer_type | port_channel_interfaces[].metadata.peer_type | Only used for validation and documentation purposes. |
| port_channel_interfaces[].validate_lldp | port_channel_interfaces[].metadata.validate_lldp | Only used for validation and documentation purposes. |
| port_channel_interfaces[].validate_state | port_channel_interfaces[].metadata.validate_state | Only used for validation and documentation purposes. |
| radius_server.hosts | radius_server.servers | Used to configure hosts with default VRF. |
| router_bgp.neighbor_interfaces[].peer | router_bgp.neighbor_interfaces[].metadata.peer | Only used for validation and documentation purposes. |
| router_bgp.neighbors[].peer | router_bgp.neighbors[].metadata.peer | Only used for validation and documentation purposes. |
| router_bgp.peer_groups[].type | router_bgp.peer_groups[].metadata.type | Only used for validation and documentation purposes. |
| router_bgp.vlan_aware_bundles[].description | router_bgp.vlan_aware_bundles[].metadata.description | Only used for validation and documentation purposes. |
| router_bgp.vlan_aware_bundles.tenant | router_bgp.vlan_aware_bundles.metadata.tenant | Only used for validation and documentation purposes. |
| router_bgp.vlans.tenant | router_bgp.vlans.metadata.tenant | Only used for validation and documentation purposes. |
| serial_number | metadata.serial_number | Only used for validation and documentation purposes. |
| vlan_interfaces.ipv6_ospf_area | vlan_interfaces[].ipv6_ospf.process.area | - |
| vlan_interfaces[].ipv6_ospf_network_point_to_point | vlan_interfaces[].ipv6_ospf.network_point_to_point | - |
| vlan_interfaces[].tags | vlan_interfaces[].metadata.tags | Only used for validation and documentation purposes. |
| vlan_interfaces[].tenant | vlan_interfaces[].metadata.tenant | Only used for validation and documentation purposes. |
| vlan_interfaces[].type | vlan_interfaces[].metadata.type | Only used for validation and documentation purposes. |
| vlans[].tenant | vlans[].metadata.tenant | Only used for validation and documentation purposes. |
| vrfs[].tenant | vrfs[].metadata.tenant | Only used for validation and documentation purposes. |
```

For more details, see the `arista.avd.cv_deploy` [role documentation](../../ansible_collections/arista/avd/roles/cv_deploy/README.md).

> #### Deprecations on eos_designs data-models

Reviewer 3: ```suggestion
#### Deprecations on eos_designs data models
```

The following data model keys have been deprecated from eos_designs in AVD 5.7.0 and will be removed in AVD 6.0.0.

| Deprecated key | New key in AVD 6.0 | Notes for AVD 6.0 |
| -------------- | ------------------ | ----------------- |
| `avd_data_validation_mode` | - | Validation is always required. |
| `avd_eos_designs_debug` | avd_tmp_dir | AVD will output internal temporary files to `avd_tmp_dir` containing validated variables and facts. |
| `avd_eos_designs_enforce_duplication_checks_across_all_models` | - | This is the builtin behavior. |

> | `avd_eos_designs_unset_facts` | - | `avd_switch_facts` are no longer set in Ansible. Can be read by setting `avd_tmp_dir`. |

Reviewer 3: @ClausHolbechArista I don't think this will be true once we start encrypting for security

Reviewer 2: I think this is ok now since we decided to use vault. @gmuloc

Reviewer 3: agreed

| `wan_use_agent_env_var_for_kernel_software_forwarding_ecmp` | `<node_type>.kernel_ecmp_cli` | Set new key to `false` to retain the behavior. |

> | `wan_use_evpn_node_settings_for_lan` | - | - |

Reviewer 4: ```suggestion
| `wan_use_evpn_node_settings_for_lan` | - | Default in AVD 6.0 |
```

!!! note
    When upgrading to AVD 6.0, refer to the porting-guide for all required changes.

> #### Deprecations on eos_cli_config_gen data-models

Reviewer 3: ```suggestion
#### Deprecations on eos_cli_config_gen data models
```

The following data model keys have been deprecated from eos_cli_config_gen in AVD 5.7.0 and will be removed in AVD 6.0.0.

| Deprecated key | New key in AVD 6.0 | Notes for AVD 6.0 |
| -------------- | ------------------ | ----------------- |
| `avd_data_validation_mode` | - | Validation is always required. |

> | `dhcp_servers[].subnets` | `dhcp_servers[].ipv4_subnets` or `dhcp_servers[].ipv6_subnets` | - |

Reviewer 4: ```suggestion
| `dhcp_servers[].subnets` | `dhcp_servers[].ipv4_subnets` or `dhcp_servers[].ipv6_subnets` | Better input data-model for IPv4 and IPv6 subnets. |
```

Author: Fixed

| `ethernet_interfaces[].peer` | `ethernet_interfaces[].metadata.peer` | Only used for validation and documentation purposes. |
| `ethernet_interfaces[].peer_interface` | `ethernet_interfaces[].metadata.peer_interface` | Only used for validation and documentation purposes. |
| `ethernet_interfaces[].peer_key` | `ethernet_interfaces[].metadata.peer_key` | Only used for validation and documentation purposes. |
| `ethernet_interfaces[].peer_type` | `ethernet_interfaces[].metadata.peer_type` | Only used for validation and documentation purposes. |
| `ethernet_interfaces[].port_porfile` | `ethernet_interfaces[].metadata.port_porfile` | Only used for validation and documentation purposes. |
| `ethernet_interfaces[].validate_lldp` | `ethernet_interfaces[].metadata.validate_lldp` | Only used for validation and documentation purposes. |
| `ethernet_interfaces[].validate_state` | `ethernet_interfaces[].metadata.validate_state` | Only used for validation and documentation purposes. |

> | `ip_ftp_client_source_interfaces` | `ip_ftp_client.source_interface` or `ip_ftp_client.vrfs[].source_interface` | - |

Reviewer 4: ```suggestion
| `ip_ftp_client_source_interfaces` | `ip_ftp_client.source_interface` or `ip_ftp_client.vrfs[].source_interface` | Better input data-model for cli order |
```
Similar for other vrf changed models

Author: Added

| `port_channel_interfaces[].peer_type` | `port_channel_interfaces[].metadata.peer_type` | Only used for validation and documentation purposes. |
| `port_channel_interfaces[].validate_lldp` | `port_channel_interfaces[].metadata.validate_lldp` | Only used for validation and documentation purposes. |
| `port_channel_interfaces[].validate_state` | `port_channel_interfaces[].metadata.validate_state` | Only used for validation and documentation purposes. |
| `radius_server.hosts` | `radius_server.servers` | Used to configure hosts with default VRF |
| `router_bgp.neighbor_interfaces[].peer` | `router_bgp.neighbor_interfaces[].metadata.peer` | Only used for validation and documentation purposes. |
| `router_bgp.neighbors[].peer` | `router_bgp.neighbors[].metadata.peer` | Only used for validation and documentation purposes. |
| `router_bgp.peer_groups[].type` | `router_bgp.peer_groups[].metadata.type` | Only used for validation and documentation purposes. |
| `router_bgp.vlan_aware_bundles[].description` | `router_bgp.vlan_aware_bundles[].metadata.description` | Only used for validation and documentation purposes. |

> | `router_bgp.vlan_aware_bundles.tenant` | `router_bgp.vlan_aware_bundles.metadata.tenant` | Only used for validation and documentation purposes. |

Reviewer 3: we have merged a PR recently where in AVD 6.0 the keys are now `tenants` and not `tenant`
```suggestion
| `router_bgp.vlan_aware_bundles.tenant` | `router_bgp.vlan_aware_bundles.metadata.tenants` | Only used for validation and documentation purposes. |
```

(same for the others below)

| `router_bgp.vlans.tenant` | `router_bgp.vlans.metadata.tenant` | Only used for validation and documentation purposes. |
| `serial_number` | `metadata.serial_number` | Only used for validation and documentation purposes. |
| `vlan_interfaces.ipv6_ospf_area` | `vlan_interfaces[].ipv6_ospf.process.area` | - |
| `vlan_interfaces[].ipv6_ospf_network_point_to_point` | `vlan_interfaces[].ipv6_ospf.network_point_to_point` | - |
| `vlan_interfaces[].tags` | `vlan_interfaces[].metadata.tags` | Only used for validation and documentation purposes. |
| `vlan_interfaces[].tenant` | `vlan_interfaces[].metadata.tenant` | Only used for validation and documentation purposes. |
| `vlan_interfaces[].type` | `vlan_interfaces[].metadata.type` | Only used for validation and documentation purposes. |
| `vlans[].tenant` | `vlans[].metadata.tenant` | Only used for validation and documentation purposes. |

> | `vrfs[].tenant` | `vrfs[].metadata.tenant` | Only used for validation and documentation purposes. |

Author: ```suggestion
| `vlans[].tenant` | `vlans[].metadata.tenants` | Only used for validation and documentation purposes. |
| `vrfs[].tenant` | `vrfs[].metadata.tenants` | Only used for validation and documentation purposes. |
```

For more details, see the `arista.avd.cv_deploy` [role documentation](../../ansible_collections/arista/avd/roles/cv_deploy/README.md).

> #### Deprecations on eos_designs data models

Reviewer 3: @Shivani-gslab this should be moved under 5.7.3 release notes as well please! :)

Author: Moved
